src/background.js

```diff
@@ -28,6 +33,32 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    //TO-DO Add method to handle indexing of bookmark
+    //bookmarkInfo.url gives the URL of the bookmark
+    //id is the unique id of the bookmark
+    //https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/bookmarks/onCreated
+    //Check if there already exists a page with this URL.
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc);
+    const reusablePage = await tryReidentifyPage({null, bookmarkInfo.url, samePageCandidates})
+    const pageData = fetchPageData(bookmarkInfo.url)
+    const favIconAttachment = formatFavIconAttachment(pageData.favIconURI)
+    const freezeDryBlob = new Blob([pageData.freezeDryHTML], {type: 'text/html;charset=UTF-8'})
+    const pageDoc = {
+        url: bookmarkInfo.url,
+        content: pageData.content,
+        _id: getPageId(pageData),
+        _attachments: {
+            'favIcon': favIconAttachment,
+            'frozen-page.html': freezeDryBlob
+        }
+    }
+    const bookmarkDoc = transformToBookmarkDoc(pageDoc, bookmarkInfo)
+    db.put(bookmarkDoc);
+    db.put(pageDoc);
```

Comment:
If you have multiple docs to insert, just use `db.bulkDocs([pageDoc, bookmarkDoc])` and leave scheduling to Pouch. As these calls return `Promise`, putting those two calls without `await` queues both doc insert `Promise`s on the event loop around the same time.

Overall the general flow of what happens on bookmark create, and your understanding of how all the different parts works seems fine! These are just little things.